script/sign.js

```diff
@@ -0,0 +1,98 @@
+// 에러 메시지 생성 함수
+function createErrorMessage(text) {
+  const errorMessage = document.createElement('div');
+  errorMessage.className = 'error-message text-medium';
+  errorMessage.innerText = text;
+  return errorMessage;
+}
+
+// 이메일 형식 검증 함수
+function isEmailValid(email) {
+  const pattern = /^[A-Za-z0-9_\.\-]+@[A-Za-z0-9\-]+\.[A-za-z0-9\-]+/;
+  return pattern.test(email);
+}
+
+document.addEventListener('DOMContentLoaded', function () {
+  const emailInput = document.getElementById('email');
+  const passwordInput = document.getElementById('password');
+  const signInButton = document.querySelector('.cta');
+  const watchPassword = document.querySelector('.eye-button');
+
+  const emailErrorMessage = createErrorMessage('');
+  emailErrorMessage.style.dsiplay = 'none';
+  const passwordErrorMessage = createErrorMessage('');
+  passwordErrorMessage.style.dsiplay = 'none';
+  emailInput.after(emailErrorMessage);
+  passwordInput.after(passwordErrorMessage);
+
+  emailInput.addEventListener('focusin', function () {
+    emailErrorMessage.style.display = 'none';
+    emailErrorMessage.classList.remove('error-box');
+  });
+
+  emailInput.addEventListener('focusout', function () {
+    const emailValue = emailInput.value.trim();
+    if (!emailValue) {
+      emailInput.classList.add('error-box');
+      emailErrorMessage.style.display = 'block';
+      emailErrorMessage.innerText = '이메일을 입력해 주세요.';
+      return;
+    }
+    if (!isEmailValid(emailValue)) {
+      emailInput.classList.add('error-box');
+      emailErrorMessage.style.display = 'block';
+      emailErrorMessage.innerText = '이메일을 확인해 주세요';
+      return;
+    }
+    emailInput.classList.remove('error-box');
+  });
+
+  passwordInput.addEventListener('focusin', function () {
+    passwordErrorMessage.style.display = 'none';
+    passwordInput.classList.remove('error-box');
+  });
+
+  passwordInput.addEventListener('focusout', function () {
+    const passwordValue = passwordInput.value;
+    if (!passwordValue) {
+      passwordInput.classList.add('error-box');
+      passwordErrorMessage.style.display = 'block';
+      passwordErrorMessage.innerText = '비밀번호 입력해주세용';
+      return;
+    }
+  });
+
+  signInButton.addEventListener('click', function (event) {
+    event.preventDefault();
+    emailInput.blur();
+    passwordInput.blur();
+    const emailValue = emailInput.value.trim();
+    const passwordValue = passwordInput.value.trim();
+
+    if (emailValue === 'test@codeit.com' && passwordValue === 'codeit101') {
+      window.location.href = '../folder.html';
+    } else {
+      emailErrorMessage.innerText = '이메일을 확인 해주세요.';
+      passwordErrorMessage.innerText = '비밀번호를 확인 해주세요.';
+      emailErrorMessage.style.display = 'block';
+      passwordErrorMessage.style.display = 'block';
+      emailInput.classList.add('error-box');
+      passwordInput.classList.add('error-box');
+    }

```

Comment:
if-else 문을 불가피하게 사용해야하는 경우를 제외하고는 사용하지 않는 것이 가독성 측면에서 더 좋습니다. else 문 대신 [early return](https://devscb.tistory.com/130)을 사용해보시면 좋습니다!
```suggestion
    if (emailValue === 'test@codeit.com' && passwordValue === 'codeit101') {
      window.location.href = '../folder.html';
      return;
    } 
      emailErrorMessage.innerText = '이메일을 확인 해주세요.';
      passwordErrorMessage.innerText = '비밀번호를 확인 해주세요.';
      emailErrorMessage.style.display = 'block';
      passwordErrorMessage.style.display = 'block';
      emailInput.classList.add('error-box');
      passwordInput.classList.add('error-box');

```

Reply:
[7115d08](https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission/pull/270/commits/7115d08e325a58000d657298bb83f42cf52933e1) 적용 완료했습니다!